kalite/topic_tools/__init__.py

```diff
@@ -67,6 +78,20 @@ def get_topic_tree(force=False, annotate=False, channel=None, language=None, par
         TOPICS = {}
     if TOPICS.get(channel) is None:
         TOPICS[channel] = {}
+
+    if settings.DO_NOT_RELOAD_CONTENT_CACHE_AT_STARTUP and not force:
```

Comment:
Yeah, I see what you mean. But now you will call `softload_json` every time `force` is `False` (if this setting is enabled). Seems like the intended behavior of the original version is something like this:

1. If `TOPICS[channel][language]` doesn't exist, then load the _un_-annotated version from disk, then annotate it.
2. If you must annotate the `TOPICS[channel][language]`, under certain conditions (`settings.DO_NOT_RELOAD_CONTENT_CACHE_AT_STARTUP and not force`) first check if the annotated version is cached, and load the cache if it is.
3. In all other cases, do something else.

In order to preserve this behavior but cut out the unnecessary `softload_json` that you identified, I think it should be like this:

```python
def get_topic_tree(force=False, annotate=False, channel=None, language=None, parent=None):
    
    # Hardcode the Brazilian Portuguese mapping that only the central server knows about
    # TODO(jamalex): BURN IT ALL DOWN!
    if language == "pt-BR":
        language = "pt"

    if not channel:
        channel = django_settings.CHANNEL

    if not language:
        language = django_settings.LANGUAGE_CODE

    global TOPICS
    if not TOPICS:
        TOPICS = {}
    if TOPICS.get(channel) is None:
        TOPICS[channel] = {}
    
    if annotate or TOPICS.get(channel, {}).get(language) is None:
        if settings.DO_NOT_RELOAD_CONTENT_CACHE_AT_STARTUP and not force:
            cached_topics = softload_json(
                cache_file_path("topic_{0}_{1}.json".format(channel, language)),
                logger=logging.debug,
                raises=False
            )
        if cached_topics:
            TOPICS[channel][language] = cached_topics
            annotate = False
            # Get rid of this and let it drop down to similar statement below
            #if parent:
            #    return filter(lambda x: x.get("parent") == parent, cached_topics)
            #else:
            #    return cached_topics
        else:
            topics = softload_json(settings.TOPICS_FILEPATHS.get(channel), logger=logging.debug, raises=False)
            # Just loaded from disk, so have to restamp.
            annotate = True

    if annotate:
        flat_topic_tree = []

        # Loop through all the nodes in the topic tree
        # and cross reference with the content_cache to check availability.
        content_cache = get_content_cache(language=language)
        exercise_cache = get_exercise_cache(language=language)
```